File: Algorithm_LevelOne/Move_Zeroes/Move_Zeroes.cc
// 283. Move Zeroes C++
#include <vector>

class Solution {
public:
    void moveZeroes(std::vector<int>& nums) {
        int count = 0;
        int size = nums.size();
        for (int i = 0; i < size; i++) {
            if (nums[i] == 0) {
                count++;
            } else if (count > 0){
                nums[i-count] = nums[i];
                nums[i] = 0;
            }
        }
    }
};

File: Algorithm_LevelOne/Move_Zeroes/Move_Zeroes.py
# 283. Move Zeroes Python

class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        count = 0
        length = len(nums)
        for i in range(length):
            if nums[i] != 0:
                nums[i-count] = nums[i]
            else:
                count+=1
        for i in range(length - count, length):
            nums[i] = 0
